src/https_client.c

```diff
@@ -98,6 +98,89 @@ static int closesocket_callback(void __attribute__((unused)) *clientp, curl_sock
   return 0;
 }
 
+static void https_log_data(enum LogSeverity level, char *ptr, size_t size)
+{
+  const size_t width = 0x10;
+
+  for (size_t i = 0; i < size; i += width) {
+    char hex[3 * width + 1];
+    char str[width + 1];
+    size_t hex_off = 0;
+    size_t str_off = 0;
+    memset(hex, 0, sizeof(hex));
+    memset(str, 0, sizeof(str));
+
+    for (size_t c = 0; c < width; c++) {
+      if (i+c < size) {
+        hex_off += snprintf(hex + hex_off, sizeof(hex) - hex_off,
+                            "%02x ", (unsigned char)ptr[i+c]);
+        str_off += snprintf(str + str_off, sizeof(str) - str_off,
+                            "%c", isprint(ptr[i+c]) ? ptr[i+c] : '.');
+      } else {
+        hex_off += snprintf(hex + hex_off, sizeof(hex) - hex_off, "   ");
+      }
+    }
+
+    LOG(level, "%4.4lx: %s%s", (long)i, hex, str);
+  }
+}
+
+static int https_curl_debug(CURL * __attribute__((unused)) handle, curl_infotype type,
+                            char *data, size_t size, void *userp)
+{
+  struct https_fetch_ctx *ctx = (struct https_fetch_ctx *)userp;
+  const char *prefix = "";
+
+  switch (type) {
+    case CURLINFO_TEXT:
+      prefix = "* ";
+      break;
+    case CURLINFO_HEADER_OUT:
+      prefix = "> ";
+      break;
+    case CURLINFO_HEADER_IN:
+      prefix = "< ";
+      break;
+    // not dumping DNS packets because of privacy
+    case CURLINFO_DATA_OUT:
+    case CURLINFO_DATA_IN:
```

Comment:
Is this overkill? 
Recompiling the code will be beyond the technical abilities of some users.
Logs are machine local and debug logs require a bunch of -v.

Reply:
Yes it may be...
But from other view:

1. I wanted to have parity with the logging we had for now.
Since that also has skipped logging of the sent/received DNS packets.

2. I think, that dumping the DNS packets are not so informative.
I would recommend to have and extra log level to print it :D (like: LOG_DEBUG_DNS)

With this change I just had prepared the stage for the next commit.
My goal is - with the 2 commit combined - to distinguish parallel requests while debugging.